src/main/java/edu/harvard/iq/dataverse/AuxiliaryFileServiceBean.java

```diff
@@ -97,8 +99,12 @@ public AuxiliaryFile processAuxiliaryFile(InputStream fileInputStream, DataFile
             storageIO.saveInputStreamAsAux(di, auxExtension);
             auxFile.setChecksum(FileUtil.checksumDigestToString(di.getMessageDigest().digest()));
 
-            Tika tika = new Tika();
-            auxFile.setContentType(tika.detect(storageIO.getAuxFileAsInputStream(auxExtension)));
+            if (mediaType != null) {
```

Comment:
Quick question - is `mediaType` (`formDataBodyPart.getMediaType();`) definitely going to be `null` here, if not supplied by the client? - As opposed to defaulting to something like `application/octet-stream`?

Reply:
Hmm. You're right. When I leave off `;type=WHATEVER` (the old way), `mediaType` is non-null and its `toString` method prints `application/octet-stream`.

Reply:
So it should just be `if (mediaType != null && (!mediaType.equals("application/octet-stream")))` here. We are doing something similar throughout the "normal" datafile upload process. 

Otherwise everything looks good in the PR.

Reply:
I didn't make any code changes but I tweaked the tests so I could make assertions on what happens if you leave off `;type=WHATEVER`: 787dbed359

I'm happy to make changes, of course. Right now we're saying the user can change the content type to whatever we want. We could change is such that if they supply `application/octet-stream` as a content type we'll run it though Tika and save whatever Tika says it is. That is, we could do something like this:

            if (mediaType != null && !mediaType.toString().equals("application/octet-stream")) {
                auxFile.setContentType(mediaType.toString());
            } else {
                Tika tika = new Tika();
                auxFile.setContentType(tika.detect(storageIO.getAuxFileAsInputStream(auxExtension)));
            }

Reply:
Yes, that's what I was saying above, how that "if" line should look like. Nobody would ever want to supply "application/octet-stream" as the type. That type just means "type unknown" or "type not specified".

Reply:
Yes, sorry, I didn't see that comment until I had written mine. I just pushed the change in 985b552a4b. Please take a look.

Reply:
Yeah, I think it looks good now - but I'd like @scolapasta and @qqmyers to chime in, on whether we really want to be doing this *instead*, and not *in addition* to saving the file name, as uploaded.